### PR TITLE
Upgrade to log4j2

### PR DESCRIPTION
**Pull request by drcrallen**
Update of #979 
Asynchronous logging enabled for select chatty classes:
* io.druid.client.ServerInventoryView
* io.druid.client.BatchServerInventoryView
* io.druid.curator.inventory.CuratorInventoryManager
* com.metamx.http.client.pool.ChannelResourceFactory

### CODE REVIEW

**Comment by drcrallen** [comment]
@fjy / @xvrl : Is this basically waiting to make sure 0.7.x is stable in our test env before bringing in more changes? or are there more comments?

**Comment by xvrl** [comment]
I would prefer to leave async logging disabled by default as it currently is and just give an example of how to configure it in the config file? By default I believe it is more important to have full logs in case of a crash rather than having good performance with debug logs enabled.

**Comment by drcrallen** [comment]
@xvrl : Reverted default to prior behavior and added documentation on how to modify behavior if desired.

File: common/pom.xml [hunk 1]
```diff
@@ -116,8 +116,27 @@
...
+        </dependency>
+        <!-- com.lmax.disruptor is optional in log4j-core, so we explicitly include it here -->
+        <dependency>
+            <groupId>com.lmax</groupId>
+            <artifactId>disruptor</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j.adapters</groupId>
+            <artifactId>log4j-1.2-api</artifactId>
+            <version>2.0-beta4</version>
```
Comment (drcrallen) [inline common/pom.xml]: THis should not be here, removing it...

**Comment by drcrallen** [comment]
I did move some comments from INFO --> DEBUG in some of the chatty classes, @xvrl : would you rather have them back to INFO?

File: server/src/main/java/io/druid/client/ServerInventoryView.java [hunk 2]
```diff
@@ -136,13 +136,13 @@ public InventoryType deserializeInventory(byte[] bytes)
           @Override
           public void newContainer(DruidServer container)
           {
-            log.info("New Server[%s]", container);
+            log.debug("New Server[%s]", container);
```
Comment (xvrl) [inline server/src/main/java/io/druid/client/ServerInventoryView.java]: sever appear/disappear are probably useful to keep as info

File: server/src/main/java/io/druid/client/ServerInventoryView.java [hunk 3]
```diff
@@ -136,13 +136,13 @@ public InventoryType deserializeInventory(byte[] bytes)
...
           }
 
           @Override
           public void deadContainer(DruidServer deadContainer)
           {
-            log.info("Server Disappeared[%s]", deadContainer);
+            log.debug("Server Disappeared[%s]", deadContainer);
```
Comment (xvrl) [inline server/src/main/java/io/druid/client/ServerInventoryView.java]: keep as info

File: server/src/main/java/io/druid/client/ServerInventoryView.java [hunk 4]
```diff
@@ -179,7 +179,7 @@ public DruidServer removeInventory(final DruidServer container, String inventory
           @Override
           public void inventoryInitialized()
           {
-            log.info("Inventory Initialized");
+            log.debug("Inventory Initialized");
```
Comment (xvrl) [inline server/src/main/java/io/druid/client/ServerInventoryView.java]: leave as info

File: extensions/s3-extensions/pom.xml [hunk 5]
```diff
@@ -96,6 +96,17 @@
...
                 </executions>
             </plugin>
+            <!-- Must set timezone to UTC to run unit tests for this module independently -->
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <!-- locale settings must be set on the command line before startup -->
+                    <argLine>-Duser.language=en -Duser.country=US</argLine>
+                    <systemPropertyVariables>
+                        <user.timezone>UTC</user.timezone>
```
Comment (xvrl) [inline extensions/s3-extensions/pom.xml]: the master pom should be taking care of UTC already

File: extensions/s3-extensions/pom.xml [hunk 6]
```diff
@@ -96,6 +96,17 @@
                     </execution>
                 </executions>
             </plugin>
+            <!-- Must set timezone to UTC to run unit tests for this module independently -->
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <!-- locale settings must be set on the command line before startup -->
+                    <argLine>-Duser.language=en -Duser.country=US</argLine>
```
Comment (xvrl) [inline extensions/s3-extensions/pom.xml]: do we know what might be relying on locale specific tests?

File: server/src/main/java/io/druid/client/ServerInventoryView.java (hunk 2 above)
Comment (drcrallen) [inline server/src/main/java/io/druid/client/ServerInventoryView.java]: fixed

File: server/src/main/java/io/druid/client/ServerInventoryView.java (hunk 3 above)
Comment (drcrallen) [inline server/src/main/java/io/druid/client/ServerInventoryView.java]: fixed

File: server/src/main/java/io/druid/client/ServerInventoryView.java (hunk 4 above)
Comment (drcrallen) [inline server/src/main/java/io/druid/client/ServerInventoryView.java]: fixed

**Pull request merged**